### ISSUE TITLE
Error - "Can't get information about the picture" for product with thumbnail image

### ISSUE THREAD

**Issue by manojVivek** [magento/magento2#17121]
We are getting the following error while accessing the checkout page. This specifically happens when the currently added product has a cross/up-selling product with a thumbnail image.

`a:4:{i:0;s:171:"Can't get information about the picture: /var/www/html/pub/media/catalog/product/cache/thumbnail/276x276/beff4985b56e3afdbeabfc89641a4582/d/t/sku.jpg";i:1;s:7993:"#0 /var/www/html/vendor/magento/module-catalog/Helper/Image.php(541): Magento\Catalog\Model\Product\Image->getResizedImageInfo()
`

But there is a valid file in the path and the permissions were correct and no problem.

### Preconditions
1. Magento version - 2.1.13
2. Php - 7.0.30
3. MySQL - AWS RDS

### Steps to reproduce
1. Make sure there is a product A with a thumbnail image in the "images and videos" section
2. For a product B, add product A as a cross-selling product.
3. Now, goto the store and add the product B to cart and checkout.
4. You will be shown the "There has been an error processing your request" page with the message:
`Can't get information about the picture: /var/www/html/pub/media/catalog/product/cache/thumbnail/276x276/beff4985b56e3afdbeabfc89641a4582/d/t/sku.jpg`

Now do the following and we will have this page load correctly:
1. Remove the thumbnail image from the Product A, in the admin dashboard and save it.
2. For a product B, add product A as a cross-selling product.
3. Now, goto the store and add the product B to cart and checkout.
4. It will be successfully taken to the checkout page.

Can someone help with fixing this issue?

**Comment by magento-engcom-team** [comment]
Hi @manojVivek. Thank you for your report. 
To help us process this issue please make sure that you provided the following information:
- [ ] Summary of the issue
- [ ] Information on your environment
- [ ] Steps to reproduce
- [ ] Expected and actual results

Please make sure that the issue is reproducible on the vanilla Magento instance following **Steps to reproduce**. To deploy vanilla Magento instance on our environment, please, add a comment to the issue: 

`@magento-engcom-team give me {$VERSION} instance` 

where `{$VERSION}` is [version tags](https://github.com/magento/magento2/tags) (starting from 2.2.0+) or develop branches (2.2-develop +). 
For more details, please, review the [Magento Contributor Assistant](https://github.com/magento/magento2/wiki/Magento-Contributor-Assistant) documentation.

@manojVivek do you confirm that you was able to reproduce the issue on vanilla Magento instance following steps to reproduce?
- [ ] yes
- [ ] no

**Comment by ghost** [comment]
@manojVivek, thank you for your report.
We were not able to reproduce this issue by following the steps you provided. Please provide more detailed steps to reproduce or try to reproduce this issue on a clean installation or latest release.

**Comment by ghost** [comment]
@manojVivek, we are closing this issue due to inactivity. If you'd like to update it, please reopen the issue.

**Comment by simonrl** [comment]
Please reopen.

We are using Magento 2.1.14.
It also happens for us when we show images in the compare and wishlist sidebar.

The problem ist with `Magento\Catalog\Model\Product\Image::getResizedImageInfo()`. There are 2 issues here:
```
$image = $this->_mediaDirectory->getAbsolutePath($this->_newFile);
if ($this->_mediaDirectory->isFile($image)) { [...]
```

Because `isFile()` converts the path to an absolute path internally, the resulting path is something like `/var/www/magento/var/www/magento` (instead of `var/www/magento`only once). This means that `isFile()` will always return false. We fixed it:
`if ($this->_mediaDirectory->isFile($this->_newFile)) { [...]`

Also, a few lines below, 
`throw new NotLoadInfoImageException(__('Can\'t get information about the picture: %1', $image));`

This will result in a full path disclosure to the Magento install directory which is a potential security risk. Replace it with:
`throw new NotLoadInfoImageException(__('Can\'t get information about the picture: %1', $this->_newFile));`

The error is shown in production mode only and is visible in the Chrome Devtools network panel, when a call to the customer section responds with a 400 status code. This results in the compare-sidebar not being rendered in production mode, and rendered with a placeholder instead of the correct product in developer mode.

**Comment by ghost** [comment]
@simonrl, please open a new issue for each problem with correctly filled Preconditions, Steps to Reproduce etc.
